compiler/rustc_codegen_llvm/src/debuginfo/mod.rs

```diff
@@ -570,7 +570,12 @@ impl<'ll, 'tcx> DebugInfoMethods<'tcx> for CodegenCx<'ll, 'tcx> {
         inlined_at: Option<&'ll DILocation>,
         span: Span,
     ) -> &'ll DILocation {
-        let DebugLoc { line, col, .. } = self.lookup_debug_loc(span.lo());
+        let (line, col) = if span == DUMMY_SP && !self.sess().target.is_like_msvc {
+            (0, 0)
```

Comment:
Discussion: I'm not too familiar with debuginfo, but this condition seems a bit sus to me -- are non-msvc-like platforms necessarily DWARF platforms? I would've expected something more like a `target.is_dwarf()` check (how that detection/definition is implemented, I unfortunately don't know).

Reply:
There is no `target.is_dwarf()`. The options are `target.is_like_msvc` which does mean "not DWARF" at the moment or matching on `target.debuginfo_kind`.

Reply:
Hm, I wonder if we should be introducing some explicit `is_dwarf()` check and not rely on the implicit `is_like_msvc` == not dwarf assumption. But I'll leave that to the reviewers more knowledgeable about debuginfo to decide.

Reply:
If people want to do that it belongs in another PR where it can be done across the board. There are several existing sites in the debuginfo code that use `is_like_msvc` this way.